Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -31,15 +31,91 @@ namespace AzToolsFramework
         }
     }
 
-    DPERowWidget::DPERowWidget(QWidget* parentWidget)
+    QSize DPELayout::sizeHint() const
+    {
+        int cumulativeWidth = 0;
+        int preferredHeight = 0;
+
+        // sizeHint for this horizontal layout is the sum of the preferred widths,
+        // and the maximum of the preferred heights
+        for (int layoutIndex = 0; layoutIndex < count(); ++layoutIndex)
+        {
+            QWidget* widgetChild = itemAt(layoutIndex)->widget();
```

Comment:
Is there a reason we grab the widget instead of calling `QLayoutItem::sizeHint`?

Reply:
I was originally planning to ignore any accidental non-widget entries, but it doesn't really matter. Fixed.